EIPS/eip-nfr.md

```diff
@@ -0,0 +1,395 @@
+
+Additionally, the ERC721 `_checkOnERC721Received` function MAY be explicitly called after mints and transfers if the smart contract aims to have safe transfers and mints.
+
+### Safe Transfers
+
+If the wallet/broker/auction application will accept safe transfers, then it MUST implement the ERC721 wallet interface.
+
+### Future Rewards `transferFrom` Function
+
+The FR `transferFrom` function MUST be called by all nFR-supporting smart contracts, though the accommodations for non-nFR-supporting contracts MAY also be implemented to ensure backwards compatibility.
+
+```solidity
+
+function transferFrom(address from, address to, uint256 tokenId, uint256 soldPrice) public virtual override payable {
+       //...
+}
+
+```
+
+The FR `transferFrom` function MUST be payable and the amount the NFT sold for MUST match the `msg.value` provided to the function. This is to ensure the values are valid and will also allow for the necessary FR to be held in the contract. Based on the stored `lastSoldPrice`, the smart contract will determine whether the sale was profitable after calling the ERC721 transfer function and transferring the NFT. If it was not profitable, the smart contract SHALL update the last sold price for the corresponding Token ID, increment the owner amount, shift the generations, and return all of the  `msg.value` to the `msg.sender` or `tx.origin` depending on the implementation. Otherwise, if the transaction was profitable, the smart contract SHALL call the `_distributeFR` function, then update the `lastSoldPrice`, increment the owner amount, and finally shift generations. The `_distributeFR` function MUST return the difference between the allocated FR that is to be distributed amongst the `_addressesInFR` and the `msg.value` to the `msg.sender` or `tx.origin`.
+
+### Future Rewards Calculation
+
+Marketplaces that support this standard MAY implement various methods of calculating or transferring Future Rewards to the previous owners.
+
+```solidity
+
+function _calculateFR(uint256 totalProfit, uint256 buyerReward, uint256 successiveRatio, uint256 ownerAmount, uint256 windowSize) pure internal virtual returns(uint256[] memory) {
+    //...        
+}
+
+```
+
+In this example (*Figure 1*), a seller is REQUIRED to share a portion of their net profit with 10 previous holders of the token. Future Rewards will also be paid to the same seller as the value of the token increases from up to 10 subsequent owners. 
+
+When an owner loses money during their holding period, they MUST NOT be obligated to share Future Rewards distributions, since there is no profit to share. However, he SHALL still receive a share of Future Rewards distributions from future generations of owners, if they are profitable.
+
+![Figure 1: Geometric sequence distribution](https://raw.githubusercontent.com/dRadiant/EIPs/master/assets/eip-nfr/Total_FR_Payout_Distribution-geo.png) 
```

Comment:
```suggestion
![Figure 1: Geometric sequence distribution](../assets/eip-5173/Total_FR_Payout_Distribution-geo.png) 
```
All links should be relative, and also don't forget to update the assets folder to `eip-5173` for all of these.